### PR TITLE
service support https access

### PR DESCRIPTION
**Pull request by shzcore**

### CODE REVIEW

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 1]
```diff
@@ -212,4 +212,27 @@ public static synchronized ServerOptions instance() {
                     disallowEmpty(),
                     "hugegraph:9fd95c9c-711b-415b-b85f-d4df46ba5c31"
             );
+    public static final ConfigOption<String> KEYSTORE_SERVER_FILE =
+            new ConfigOption<>(
+                    "server.keystore.file",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:217]: prefer ssl.server_keystore_file

File: hugegraph-api/src/main/java/com/baidu/hugegraph/server/RestServer.java [hunk 2]
```diff
@@ -68,8 +70,22 @@ public void start() throws IOException {
...
+        String protocol = this.conf.get(ServerOptions.SECURITY_PROTOCOL);
+        String keystoreServerFile = this.conf.get(ServerOptions.KEYSTORE_SERVER_FILE);
+        String keystoreServerPassword = this.conf.get(ServerOptions.KEYSTORE_SERVER_PASSWORD);
+        if (protocol != null && protocol.equals("https")) {
+            SSLContextConfigurator sslContext = new SSLContextConfigurator();
+            // set up security context
+            sslContext.setKeyStoreFile(keystoreServerFile);
+            sslContext.setKeyStorePass(keystoreServerPassword);
+            server = GrizzlyHttpServerFactory.createHttpServer(uri, rc, true,
+                                                               new SSLEngineConfigurator(sslContext)
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/server/RestServer.java:83]: define a var for 'new SSLEngineConfigurator(sslContext)'

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 3]
```diff
@@ -212,4 +212,27 @@ public static synchronized ServerOptions instance() {
...
+            new ConfigOption<>(
+                    "server.keystore.password",
+                    "The server keystore password.",
+                    disallowEmpty(),
+                    "abc123"
+            );
+
+    public static final ConfigOption<String> SECURITY_PROTOCOL =
+            new ConfigOption<>(
+                    "security.protocol",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:233]: prefer server.protocol

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 4]
```diff
@@ -212,4 +212,27 @@ public static synchronized ServerOptions instance() {
...
+                    disallowEmpty(),
+                    "server.keystore"
+            );
+
+    public static final ConfigOption<String> KEYSTORE_SERVER_PASSWORD =
+            new ConfigOption<>(
+                    "server.keystore.password",
+                    "The server keystore password.",
+                    disallowEmpty(),
+                    "abc123"
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:228]: don't set by default, just keep empty

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 5]
```diff
@@ -212,4 +212,27 @@ public static synchronized ServerOptions instance() {
                     disallowEmpty(),
                     "hugegraph:9fd95c9c-711b-415b-b85f-d4df46ba5c31"
             );
+    public static final ConfigOption<String> KEYSTORE_SERVER_FILE =
+            new ConfigOption<>(
+                    "server.keystore.file",
+                    "The server keystore file.",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:218]: improve description: The path of server keystore file used when https protocol is enabled

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 6]
```diff
@@ -212,4 +212,27 @@ public static synchronized ServerOptions instance() {
...
+                    disallowEmpty(),
+                    "server.keystore"
+            );
+
+    public static final ConfigOption<String> KEYSTORE_SERVER_PASSWORD =
+            new ConfigOption<>(
+                    "server.keystore.password",
+                    "The server keystore password.",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:226]: improve description

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 7]
```diff
@@ -212,4 +212,27 @@ public static synchronized ServerOptions instance() {
...
+            new ConfigOption<>(
+                    "server.keystore.file",
+                    "The server keystore file.",
+                    disallowEmpty(),
+                    "server.keystore"
+            );
+
+    public static final ConfigOption<String> KEYSTORE_SERVER_PASSWORD =
+            new ConfigOption<>(
+                    "server.keystore.password",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:225]: prefer ssl.server_keystore_password

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 8]
```diff
@@ -212,4 +212,27 @@ public static synchronized ServerOptions instance() {
...
+                    "The server keystore password.",
+                    disallowEmpty(),
+                    "abc123"
+            );
+
+    public static final ConfigOption<String> SECURITY_PROTOCOL =
+            new ConfigOption<>(
+                    "security.protocol",
+                    "security.protocol.",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:234]: improve description

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 9]
```diff
@@ -212,26 +212,29 @@ public static synchronized ServerOptions instance() {
...
             );
-    public static final ConfigOption<String> KEYSTORE_SERVER_FILE =
+
+    public static final ConfigOption<String> SERVER_KEYSTORE_FILE =
             new ConfigOption<>(
-                    "server.keystore.file",
-                    "The server keystore file.",
+                    "ssl.server_keystore_file",
+                    "The path of server keystore file used when https " +
+                            "protocol is enabled.",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:220]: align with "The path..."

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 10]
```diff
@@ -212,26 +212,29 @@ public static synchronized ServerOptions instance() {
...
-    public static final ConfigOption<String> KEYSTORE_SERVER_PASSWORD =
+    public static final ConfigOption<String> SERVER_KEYSTORE_PASSWORD =
             new ConfigOption<>(
-                    "server.keystore.password",
-                    "The server keystore password.",
-                    disallowEmpty(),
-                    "abc123"
+                    "ssl.server_keystore_password",
+                    "The password of the path of the server keystore file " +
+                            "used when the https protocol is enabled.",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:229]: ditto

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 11]
```diff
@@ -212,26 +212,29 @@ public static synchronized ServerOptions instance() {
...
+                    ""
             );
 
-    public static final ConfigOption<String> SECURITY_PROTOCOL =
+    public static final ConfigOption<String> SERVER_PROTOCOL =
             new ConfigOption<>(
-                    "security.protocol",
-                    "security.protocol.",
+                    "server.protocol",
+                    "Fill in when https protocol is enabled.",
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:237]: The protocol of rest-server, allowed values are: http or https.

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 12]
```diff
@@ -212,4 +212,30 @@ public static synchronized ServerOptions instance() {
...
+                            "used when the https protocol is enabled.",
+                    null,
+                    ""
+            );
+
+    public static final ConfigOption<String> SERVER_PROTOCOL =
+            new ConfigOption<>(
+                    "server.protocol",
+                    "Fill in when https protocol is enabled.",
+                    disallowEmpty(),
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:238]: allowValues("http", "https")

File: hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java [hunk 13]
```diff
@@ -212,4 +213,31 @@ public static synchronized ServerOptions instance() {
...
+                    null,
+                    ""
+            );
+
+    public static final ConfigOption<String> SERVER_PROTOCOL =
+            new ConfigOption<>(
+                    "server.protocol",
+                    "The protocol of rest-server, allowed values are: " +
+                    "http or https.",
+                    allowValues("http","https"),
```
Comment (Linary) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/config/ServerOptions.java:240]: add an space between "http" and "https"

**Pull request merged**